### ISSUE TITLE
[feature][boyb] review empty values

### ISSUE THREAD

**Issue by laurentsimon** [slsa-framework/slsa-github-generator#1512]
In the TS code, we have been using 
```ts
val = <some-var> || ""
```

Most of the time, these env variables should never be empty because they are GitHub variables. Nevertheless, we should spend some time reviewing all the occurrences of such code and verify they are fine. To ease code review, we should maybe *always* validate that they are *not* empty, and return an error if they are not.

**Comment by asraa** [comment]
yes! in https://github.com/slsa-framework/slsa-github-generator/pull/1477 i added a function that gets from env var and throws if not defined. 

but it's only for the critical ones. i haven't audited which ones we should enforce otherwise.

**Comment by ianlewis** [comment]
It might be best for us to check these values in e2e tests? I'm not sure we always need to produce an error on the generator side, but we should try to make sure we are capturing the values properly and don't have bugs/typos etc.

**Comment by ianlewis** [comment]
Also, part of the reason we need to do this is that env vars have the TypeScript type of `string | undefined` so if we want to pass them to functions that take strings we need to convert them.
